### DISCUSSION TITLE
How can we have a bot run a release workflow that does pre-release changes?

### DISCUSSION THREAD

**Discussion by mieubrisse** [community/community#37547] [Repositories]
We're trying to set up a safe and simple workflow for devs where:
1. Branch protections are on, devs can only merge to `master` via PR, and those PRs must have CI checks pass
2. Each PR must have a changelog entry, so we're building a user-friendly changelog with each PR
3. Devs can cut releases using a manual workflow, and cutting a release changes the `TBD` section of queued changelog entries to the correct `X.Y.Z` version being released

However, we're struggling to make this happen with current settings.

Our current situation is as follows:
1. We have a `master` branch
1. Release versions are denoted as semver `X.Y.Z` tags on the master branch
1. Branch protections are on, requiring PRs, but a bot user account is given permission to bypass the PR requirement (important)
1. Each PR has a CI check to ensure the dev has updated the changelog
1. Devs PR their features into `master`
1. Our Github org has a secret containing an access token that allows behaving as the bot
1. When it comes time to release, the dev specifies the `X.Y.Z` version to release and runs a manually-triggered GH Action:
    1. Uses the bot access token to behave as the bot (which can bypass the PR requirement for committing to `master`)
    2. Updates the `TBD` section of the changelog to `X.Y.Z` in a commit
    3. Pushes the commit to `master`
    4. Tags the commit with `X.Y.Z`

The struggles we're having with the current situation:
1. We can't turn on "require status checks" because the bot will no longer be able to push the changelog updates to the `master` branch. This means that we can't turn on "enable auto-merge", else auto-merge will merge PRs with failing CI checks.
2. We can't turn on "Don't allow bypassing" (which would prevent our devs from pushing to `master` by accident) because then the bot would be prevented from committing the changelog updates to the `master` branch
3. We need to have the "god" bot account, which has push access everywhere
4. We need to have the token allowing the releasing user to act as the bot as an org-wide secret

How can we do this with Github?

Suggestion: it _seems_ like a "allow these users to bypass all branch protection (CI checks, PR, etc.)" whitelist would fix this problem (currently we can only bypass the PR requirement)

**Comment by ossanna16** [comment]
Hi there @mieubrisse and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
